src/components/cavatica/CavaticaAddProject.js

```diff
@@ -56,84 +57,140 @@ const AddIcon = styled(PlusIcon)`
 const enhance = compose(
   injectState,
   withTheme,
-  withState('projectName', 'setProjectName', ''),
-  withState('addingProject', 'setAddingProject', false),
-  withState('billingGroups', 'setBillingGroups', []),
   withState('billingGroup', 'selectBillingGroup', null),
-  lifecycle({
-    async componentDidMount() {
-      const { setBillingGroups } = this.props;
-      getBillingGroups().then(bg => setBillingGroups(bg));
-    },
-  }),
 );
 
-const CavaticaAddProject = ({
-  theme,
-  projectName,
-  setProjectName,
-  addingProject,
-  setAddingProject,
-  setSelectedProject,
-  billingGroups,
-  selectedBillingGroup,
-  selectBillingGroup,
-  onSuccess,
-}) => {
-  const onSaveButtonClick = async () => {
-    saveProject({
-      projectName,
-      selectedBillingGroup,
-      billingGroups,
-    }).then(({ id }) => {
-      setAddingProject(false);
-      setProjectName('');
+const defaultState = {
+  projectName: '',
+  billingGroups: [],
+  selectedBillingGroup: null,
+  error: null,
+  addingProject: false,
+  //Do not show 'try again' if getBillingGroups fails. To show it, we should add a key to the component and change it to refresh the component. The parent is a bit too complicated to add extra logic to it so I prefer to make things simple...
+  isErrorFromBillingGroups: false,
+};
+
+class CavaticaAddProject extends Component {
+  state = {
+    ...defaultState,
+  };
+
+  async componentDidMount() {
+    try {
+      const billingGroups = (await getBillingGroups()) || [];
+      this.setState({ billingGroups });
+    } catch (error) {
+      this.setState({ error, isErrorFromBillingGroups: true });
+    }
+  }
+
+  @bind
+  async onSaveButtonClick() {
+    const { projectName, selectedBillingGroup, billingGroups } = this.state;
+    const { onSuccess, setSelectedProject } = this.props;
+
+    try {
+      this.setState({ isSaveButtonDisabled: true });
+      const { id } = await saveProject({
+        projectName,
+        selectedBillingGroup,
+        billingGroups,
+      });
       setSelectedProject(id);
       onSuccess();
-    });
-  };
-  const onProjectNameChange = e => setProjectName(e.target.value);
-  const onBillingGroupSelect = e => selectBillingGroup(e.target.value);
-  const onCancelClick = () => setAddingProject(false);
-  const onCreateButtonClick = () => setAddingProject(true);
-  return (
-    <Container>
-      {addingProject ? (
-        <Fragment>
+      this.setState({
+        projectName: '',
+        addingProject: false,
+      });
+    } catch (error) {
+      this.setState({ error });
+    } finally {
+      this.setState({ projectName: '', isSaveButtonDisabled: false });
+    }
+  }
+
+  @bind
```

Comment:
We'll have to stop using decorators as it will not be supported by CRA until the API is stable.

Generally speaking, we should refrain from using ES features that do not have a stable spec (at least stage 4).

Reply:
See https://github.com/facebook/create-react-app/issues/411